extensions/sql-database-projects/src/controllers/databaseProjectTreeViewProvider.ts

```diff
@@ -0,0 +1,120 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import * as constants from '../common/constants';
+
+import { SqlDatabaseProjectItem } from './databaseProjectTreeItem';
+
+export class SqlDatabaseProjectTreeViewProvider implements vscode.TreeDataProvider<SqlDatabaseProjectItem> {
+	private _onDidChangeTreeData: vscode.EventEmitter<SqlDatabaseProjectItem | undefined> = new vscode.EventEmitter<SqlDatabaseProjectItem | undefined>();
+	readonly onDidChangeTreeData: vscode.Event<SqlDatabaseProjectItem | undefined> = this._onDidChangeTreeData.event;
+
+	private roots: SqlDatabaseProjectItem[] = [];
+
+	constructor() {
+		this.initialize();
+	}
+
+	private initialize() {
+		this.roots = [new SqlDatabaseProjectItem(constants.noOpenProjectMessage, false)];
+	}
+
+	public getTreeItem(element: SqlDatabaseProjectItem): vscode.TreeItem {
+		return {
+			label: element.label,
+			collapsibleState: element.parent === undefined
+				? vscode.TreeItemCollapsibleState.Expanded
+				: element.isFolder
+					? vscode.TreeItemCollapsibleState.Collapsed
+					: vscode.TreeItemCollapsibleState.None
+		};
+	}
+
+	public getChildren(element?: SqlDatabaseProjectItem): SqlDatabaseProjectItem[] {
+		if (element === undefined) {
+			return this.roots;
+		}
+
+		return element.children;
+	}
+
+	public async openProject(projectFiles: vscode.Uri[]) {
```

Comment:
you don't have any watcher for when files are added to the project folders. Are you going to add that later?